cli/command/load_balancer_command_test.go

```diff
@@ -273,3 +286,100 @@ func TestListLoadBalancers(t *testing.T) {
 		t.Fatal(err)
 	}
 }
+
+func TestHealthCheck_noUpdateRequired(t *testing.T) {
+	tc, ctrl := newTestCommand(t)
+	defer ctrl.Finish()
+	command := NewLoadBalancerCommand(tc.Command())
+
+	tc.Resolver.EXPECT().
+		Resolve("load_balancer", "env").
+		Return([]string{"id"}, nil)
+
+	tc.Client.EXPECT().
+		GetLoadBalancer("id").
+		Return(&models.LoadBalancer{}, nil)
+
+	// We're testing a call is NOT made to the client for updating the health check
+	// for example:
+	// tc.Client.EXPECT_NOT().
+	//     UpdateLoadBalancerHealthCheck(gomock.Any(), gomock.Any())
+
+	c := getCLIContext(t, Args{"env", "name"}, nil)
+	if err := command.HealthCheck(c); err != nil {
+		t.Fatal(err)
+	}
+}
+
+func TestHealthCheck_partialUpdateRequired(t *testing.T) {
+	tc, ctrl := newTestCommand(t)
+	defer ctrl.Finish()
+	command := NewLoadBalancerCommand(tc.Command())
+
+	existingHealthCheck := models.HealthCheck{
+		Target:             "TCP:80",
+		Interval:           30,
+		Timeout:            5,
+		HealthyThreshold:   2,
+		UnhealthyThreshold: 2,
+	}
+
+	tc.Resolver.EXPECT().
+		Resolve("load_balancer", "env").
+		Return([]string{"id"}, nil)
+
+	tc.Client.EXPECT().
+		GetLoadBalancer("id").
+		Return(&models.LoadBalancer{
+			HealthCheck: existingHealthCheck,
+		}, nil)
+
+	expectedHealthCheck := models.HealthCheck{
+		Target:             "TCP:88",
+		Interval:           45,
+		Timeout:            5,
+		HealthyThreshold:   2,
+		UnhealthyThreshold: 2,
+	}
+
+	tc.Client.EXPECT().
+		UpdateLoadBalancerHealthCheck("id", expectedHealthCheck)
+
+	flags := Flags{
+		"set-target":   "TCP:88",
+		"set-interval": 45,
+	}
+
+	c := getCLIContext(t, Args{"env", "name"}, flags)
+	if err := command.HealthCheck(c); err != nil {
+		t.Fatal(err)
+	}
+}
+
+func TestHealthCheck_userInputErrors(t *testing.T) {
+	tc, ctrl := newTestCommand(t)
+	defer ctrl.Finish()
+	command := NewLoadBalancerCommand(tc.Command())
+
+	tc.Resolver.EXPECT().
+		Resolve("load_balancer", "env").
+		Return([]string{"id"}, nil)
+
+	tc.Client.EXPECT().
+		GetLoadBalancer("id").
+		Return(&models.LoadBalancer{}, nil)
+
+	flags := Flags{
+		"set-target": "TCP",
+	}
+
+	contexts := map[string]*cli.Context{
```

Comment:
Good call on the non-int values; updating that.

The way the command is set up, though, the user doesn't need to specify all of the flags (I assume that's what you mean by missing args?) - just the ones they want to update. Calling `loadbalancer healthcheck --set-interval 5 name-of-lb` shouldn't fail even though none of the other flags are defined.

Also, since AWS has decent `target` validation errors, it seems to me that I don't need to test for it in this function. Sound right?